tests/features/layers/activation-order/test/testcase.cpp

```diff
@@ -0,0 +1,177 @@
+/* -*- mode: c++ -*-
+ * Copyright (C) 2020  Keyboard.io, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify it under
+ * the terms of the GNU General Public License as published by the Free Software
+ * Foundation, version 3.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "testing/setup-googletest.h"
+
+SETUP_GOOGLETEST();
+
+namespace kaleidoscope {
+namespace testing {
+namespace {
+
+using ::testing::IsEmpty;
+
+class LayerActivationOrder : public VirtualDeviceTest {
+ public:
+  void SingleKeyTest(Key k) {
```

Comment:
Can this get a more descriptive name?  maybe something active. 
And I know that all our tests have the key at 0,0, but maybe we want to pass it in as a param anyway.

TestPressAndRelease(KEYSWITCH_TOP_LEFT, Key_A); ?